Issue #1077: Hide Firware update menu item when using wifi.

ddimensia opened the pull request:
This may have to be revisited if the app ever supports switching connection types at runtime but for now it seems to work as intended.

brentpicasso commented on autosportlabs/racecapture/api/rcpapi.py:
```diff
@@ -758,6 +760,9 @@ def sample(self, include_meta=False):
             self.sendCommand({'s':0})
 
+    def is_firmware_update_supported(self):
+        return not is_mobile_platform() and not isinstance(self.comms, SocketComm)
```
IMO it would be good to not have rcpapi have any knowledge of specific Comms objects. I wonder if we could push that knowledge down to the individual Comms objects, or put the knowledge in the comms_factory. 

firmware upgrading can only happen on a 'direct connection' (usb). or a 'non wireless' connection. perhaps we can have an is_wireless property on each comms object and derive if firmware is update-able based on whether the connection is not wireless?

ddimensia commented:
Rewrote to rely on is_wireless method on Comms objects, dropped is_mobile_platform check as that was just a surrogate for the is_wireless check.   Updated unit tests and verified manually via OSX + RaceCapture unit.

brentpicasso commented:
Looks good. merging!
